File: 15. Divisor, multiple, decimal 2/_13909.java
import java.util.Scanner;

public class _13909 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        
        boolean[] arr = new boolean[n + 1];

        for (int i = 2; i <= n; i++) {
            int num = i;
            while (num <= n) {
                if (arr[num] == true) {
                    arr[num] = false;
                } else {
                    arr[num] = true;
                }
                num += i;
            }
        }

        int count = 0;
        for (boolean b : arr) {
            if (b == false) {
                count++;
            }
        }

        System.out.println(count - 1);
        
    }
}

File: 15. Divisor, multiple, decimal 2/_13909.py
n = int(input())

arr = [True for _ in range(n + 1)]

for i in range(2, n + 1):
    num = i
    while num <= n:
        if arr[num] == True:
            arr[num] = False
        else:
            arr[num] = True
        num += i

# bool 배열을 sum 할 경우 true는 1로 false는 0으로 계산
print(sum(arr) - 1)